strapi/strapi
# chore: upgrade deps and pinned subdeps for CVEs

### What does it do?

direct package upgrades for packages that included vulnerable packages include:

- `path-to-regexp` 6.2.1 to 6.3.0
- `lodash.set` replaced with lodash because it seems unmaintained and contains a vulnerability
- `chokidar` from 3.5.3 to 3.6.0
- `@vercel/ncc`  from 3.8.0 to 3.8.2 (required to handle the updates that happened with webpack and babel I believe)
- `prettier` from 3.2.5 to 3.3.3 universally (was mismatched between packages)
- `docusaurus` packages from 3.1.1 to 3.5.2
- `lint-staged` from 13.3.0 to 15.2.10
- `typescript` to 5.3.2 universally (was mismatched between packages)
- `@apollo/server` from 4.10.0 to 4.11.0 (no breaking changes listed in [their changelog](https://github.com/apollographql/apollo-server/blob/main/packages/server/CHANGELOG.md))
- `vite` from 5.1.7 to 5.2.14

...and then too many pinned subdependencies to list from running `yarn dedupe && yarn up -R` along with manually removing and updating packages that contained vulnerable subdependencies which for some reason weren't caught be `yarn up`

### Why is it needed?

Resolves:

CVE-2023-45133 (babel/traverse)
CVE-2024-45590 (bodyparser)
CVE-2024-21529 (dset)
CVE-2024-45801 (dompurify)
CVE-2024-45296 (path-to-regexp)
CVE-2024-4068 (braces)
CVE-2024-45812 (vite)
CVE-2024-45811 (vite) 

after this is done, we'll re-enable dependabot against `develop` so it becomes less of a challenge


The following are not being updated here, and will be handled in a later PR. Strapi itself is not actually vulnerable from anything stemming from them:

- `@strapi/eslint-config`
- `@strapi/pack-up` 
- `plop`

### How to test it?

everything should still work within the monorepo build and lint and prettier

I had to `yarn cache clean` and delete my node_modules a few times during the process, you might need to do that before building

Then actual Strapi can be tested with experimental: 0.0.0-experimental.3162602be5ecf801e9697f416d1467608d46ad32 :

`yarn create-strapi-app@0.0.0-experimental.3162602be5ecf801e9697f416d1467608d46ad32`

### Related issue(s)/PR(s)

DX-1633
